### PR TITLE
Help menu: use north-east arrow as hyperlink suffix

### PR DESCRIPTION
**Pull request by ronso0**
Restore the north-east arrow for external links.

*ubuntu with xfce
image

Windows10
image

I also considered 🌐
image
but it may look really sh*#ty on various OS, also it's colored and distracting.

I think the arrow is fine, and much better than current ` => `
Looks fine on
- [x] *ubuntu
- [x] Windows
- [x] ~~macOS~~ excluded

### CODE REVIEW

**Comment by JoergAtGithub** [comment]
A chain symbol is common for links, like here in GitHub

**Comment by ronso0** [comment]
Sure, if you refer to https://unicode-table.com/en/1F517/ 

But it's a large icon. Here in the comment box it looks like this
image
and it's rendered like this
 "Community Support 🔗" 
On my system it looks like this
image

That's why I want to include a standard char from the Dingbats set which is included in the font we will use for the main menu #3704

**Comment by ronso0** [comment]
with #3704 
image

**Comment by foss-** [comment]
macOS 11.2.3
2.3.0-beta (build HEAD r8079)
If possible, remove arrow for macOS only as it is not used in menus.

help menu

**Comment by ronso0** [comment]
but the `=>` was shown in macOS previously, wasn't it?

**Comment by Holzhaus** [comment]
How about this icon? https://www.fileformat.info/info/unicode/char/1f310/index.htm

**Comment by ronso0** [comment]
(getting a server error) 
my point here is just to get rid of those ugly characters which were introduced back then because windows couldn't display a hyperlink character.
I do not really invest work to append an SVG and test if that is displayed properly on all os.
the char added here is working on all os, and I can not show it macOS.

**Comment by foss-** [comment]
> but the `=>` was shown in macOS previously, wasn't it?
Y

**Comment by ronso0** [comment]
@Holzhaus 0x1F310 is the globe I mentioned in the first post. It's in https://unicode-table.com/de/blocks/miscellaneous-symbols-and-pictographs/ and not guaranteed to work in most standard fonts, for example it's not in Open Sans.
image
btw that's the hyperlink char ; )
> get rid of those ugly characters which were introduced back then because windows couldn't display a hyperlink character.

**Comment by Holzhaus** [comment]
This works for me. But for me it's non-obvious that the northeast arrow denotes a website.

Have you considered setting an SVG icon for the actions instead? E.g. a globe? it would be displayed on the left, but I think it would be more clear.

**Comment by ronso0** [comment]
> This works for me. But for me it's non-obvious that the northeast arrow denotes a website.

square + north-east arrow is the default for external links for on wikipedia fr example, that's why I picked it.

> Have you considered setting an SVG icon for the actions instead? E.g. a globe? it would be displayed on the left, but I think it would be more clear. Then you also have full control over the color.

nope, because IMO a globe at this size doesn't make much sense. If I look at other menu icons, in file managers for example, I find the combination of icon size and level of detail not very helpful tbh. On screens with regular dpi (96?) I find those icons rather blurry and the only thing that helps me to navigate is mostly the color, maybe the rough shape, but the details are useless IMO.
Firefox bookmark menu:
image

**Comment by ronso0** [comment]
Are there any obejctions to merge this, even if it's just a quick fix to get rid of => ?

**Comment by daschuer** [comment]
As far as I remember it was originally the ↗
We have removed it because of windows XP.
Nice that we are back now.

File: src/widget/wmainmenubar.cpp [hunk 1]
```diff
@@ -521,7 +521,10 @@ void WMainMenuBar::initialize() {
     // HELP MENU
     QMenu* pHelpMenu = new QMenu(tr("&Help"), this);
 
-    QString externalLinkSuffix = " =>";
+    QString externalLinkSuffix;
+#ifndef __APPLE__ // external links don't have icons on macOS
```
Comment (daschuer) [inline src/widget/wmainmenubar.cpp:525]: accoring to ....

File: src/widget/wmainmenubar.cpp [hunk 2]
```diff
@@ -521,7 +521,10 @@ void WMainMenuBar::initialize() {
...
+    externalLinkSuffix = " " + QChar(0x2197);
```
Comment (daschuer) [inline src/widget/wmainmenubar.cpp:526]:
```suggestion
    externalLinkSuffix = QChar(' ') + QChar(0x2197);  // north east arrow
```


```

**Comment by daschuer** [comment]
Did we have confirmation that this works on Windows?

File: src/widget/wmainmenubar.cpp (hunk 1 above)
Comment (ronso0) [reply src/widget/wmainmenubar.cpp:525]: acording to screenshots from @foss- and web search for 'macos help menu'

**Comment by ronso0** [comment]
not yet. will test it tonight myself on Win10

**Comment by daschuer** [comment]
I have found the apple user interface guidelines. 
They defin a small set of symbols. Our north west icon is not one of them.

They also want "Title Style Capitalization"
This matter for "Translate this Application" 
When at all, a 2.4 topic to not break translations.

**Comment by foss-** [comment]
Not sure we should overcomplicate this. @daschuer why are we discussing icon design for macOS? There will not be any icon shown on macOS for `Menubar > Help` items. So there really is nothing to discuss in that regard.

Mixxx uses `Title Style Capitalization` so nothing to change in that regard or am I missing something? Besides not the scope of this specific PR.

For reference here's the [Apple Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/macos/menus/menu-bar-menus/).

**Comment by daschuer** [comment]
I am fine with the look here. It is just a matter of documentation the special case code. 

The this in "Translate this Application" needs to be lower case to be fully complaint.

**Comment by esbrandt** [comment]
> As far as I remember it was originally the ↗
> We have removed it because of windows XP.
> Nice that we are back now.

The cycle is completed.  
https://github.com/mixxxdj/mixxx/commit/86689e31a7338512100275ed1302d17932383382
👍

**Comment by ronso0** [comment]
> > As far as I remember it was originally the arrow_upper_right
> > We have removed it because of windows XP.
> > Nice that we are back now.
> 
> The cycle is completed.
> [86689e3](https://github.com/mixxxdj/mixxx/commit/86689e31a7338512100275ed1302d17932383382)
> +1

**Comment by ronso0** [comment]
To wrap up the macOS topic:
@daschuer I assume you refer to 'Using Symbols in Menus' in https://developer.apple.com/design/human-interface-guidelines/macos/menus/menu-anatomy/
Indeed the arrow is not mentioned, also they encourage to "Avoid using custom symbols in menus"
Otoh we can't compare the Mixxx Help menu with other macOS apps as appearantly the Help item there is usually meant to search the app's help book.
So special case here, let's just leave not show it on macOS to not confuse anyone ; \

**Comment by ronso0** [comment]
Ready for merge..

**Comment by ronso0** [comment]
okay, so we try to have a local manual.
let's also have a local cheat sheet for keyboard shortcuts then, right? the PNG from the manual will do

**Comment by esbrandt** [comment]
> actually a PDF created from /res/images/templates/ic_template_keyboard_mapping_sheet.svg _could_ be really small but unfortunately all key labels are already converted to paths :\

A pdf generated from the template is about 300 kb in size. Too big? 

I find it rather problematic that the keyboard's labeling is En-US, which does not apply to a lot of our users. Perhaps we should simply omit the labels (except for the function keys).

**Comment by ronso0** [comment]
hmkay, when I exported it was 1.6mb.
yes, omitting the labels would work for me

**Comment by ronso0** [comment]
Alright, let's add the keyboard cheat sheet in another PR. I'll have to learn about including it in the packing process first.
@esbrandt Can you describe your PDF export process and settings? I tried with the default settings in Inkscape which resulted in the aforementioned 1.6mb

**Comment by ronso0** [comment]
So this here is ready to be merged, right?
This is how it looks like on Windows10
image

**Comment by daschuer** [comment]
Yes thank you.

**Comment by esbrandt** [comment]
> @esbrandt Can you describe your PDF export process and settings? I tried with the default settings in Inkscape which resulted in the aforementioned 1.6mb

Exporting with Inkscape results in 300 kb pdf
[ic_template_keyboard_mapping_sheet.pdf](https://github.com/mixxxdj/mixxx/files/6176272/ic_template_keyboard_mapping_sheet.pdf)

Setting used:
Capto_Capture 2021-03-20_08-18-13_PM

**Pull request merged**